bitblas/ops/general_flashatten/__init__.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+from bitblas.base.roller.hint import Hint
+from tvm.target import Target
+from .tilelang import select_scheduler as consistent_scheduler
+from ..base_scheduler import BaseScheduler
+from ..operator import OperatorConfig, Operator, BaseKernelNameGenerator
+from ...base.arch.cuda import CUDA
+from ...utils import auto_detect_nvidia_target
+from dataclasses import dataclass
+from typing import Union, Tuple, Literal, Optional
+import logging
+import torch
+
+logger = logging.getLogger(__name__)
+
+WORKSPACE_SIZE = 1024 * 1024 * 256
+
+
+def is_native_compute(Q_dtype, K_dtype, V_dtype) -> bool:
+    return Q_dtype == K_dtype and K_dtype == V_dtype
+
+
+@dataclass(frozen=True)
+class FlashAttenConfig(OperatorConfig):
+    batch: Union[int, Tuple[int]] = None
+    # TODO should distinguish from q_heads and kv_heads
+    heads: int = None
```

Comment:
recommend to be 

```
from typing import Optional
...
heads: Optional[int] = None
heads: Optional[int] = None
```

Reply:
Thanks, both flashatten config and matmul config are modified using `Optional`